[FIX] mail: move subdomain DNS setup from website to inbound mail doc (#18187)
odoo/documentation: content/applications/general/email_communication/email_servers_inbound.rst

.. image:: email_servers_inbound/custom-subdomain-sh.png
   :alt: Setting up an Odoo subdomain on a branch.

> .. note::
>    For more information on configuring DNS records for your subdomain, see
>    :ref:`DNS record setup <dns_record_setup>`.

Reviewer: ```suggestion
   For more information on configuring DNS records for subdomains, see
   :ref:`DNS record setup <dns_record_setup>`.
```

Emails sent to addresses on that subdomain are then routed directly to Odoo.

.. important::
   For Odoo Online and Odoo.sh, CNAME-based routing is only supported when using a dedicated
   subdomain. It cannot be configured on your root domain (e.g., `mydomain.com`).

.. note::

   For instructions on managing external email providers, see the
   :ref:`Use an external email provider <domain-name/external-email>` section of the
   Domain names documentation.

.. tabs::

   .. group-tab:: Odoo Online

      - Configure your DNS records
         - If your domain is hosted with a third-party registrar (for example, GoDaddy, Namecheap, Cloudflare), sign in to the registrar and open the DNS management zone for your domain.
         - If Odoo manages your DNS:
            - Open the `database manager <https://www.odoo.com/my/databases>`_.
            - Click the :icon:`fa-gear` (:guilabel:`gear`) button beside the database and choose :icon:`fa-globe` :guilabel:`Domain Names`.
            - Click :guilabel:`DNS`, then :guilabel:`Add DNS record`.

      - Create a CNAME record
         - **Name / Host:** the subdomain you want to use (for example, `mail`).

>          - **Content / Target:** your database root domain with a trailing dot (for example, `mydomain.com.`).

Reviewer 2: Since this is intended for Odoo Online customers, I would advise using an odoo.com subdomain as the example for the CNAME value --> for example, `dbname.odoo.com.`

Whether or not a trailing dot is required at the end of the FQDN depends on the provider; not all of them require it.

Author: Ok for the CNAME example

I will remove the "trailing dot" part as this is explained when adding a CNAME record on Odoo (or do you want me to add that it depends on the provider)

      - Connect the subdomain in Odoo
         - In the `database manager <https://www.odoo.com/my/databases>`_, click the :icon:`fa-gear` (:guilabel:`gear`) button for your database and select :icon:`fa-globe` :guilabel:`Domain Names`.
         - Click :guilabel:`Use my own domain`, enter the full subdomain (for example, `mail.mydomain.com`), then click :guilabel:`Verify`. When verification succeeds, click :guilabel:`I confirm, it's done`.

      - Add the alias domain inside your database
         - Log in to the database and open the :guilabel:`Settings` app.
         - In the :guilabel:`Alias Domain` field enter the subdomain you configured (for example, `mail.mydomain.com`).

         .. image:: email_servers_inbound/alias-domain.png
            :alt: Adding an alias domain in the settings.
            :scale: 50%

   .. group-tab:: Odoo.sh

      - Configure your DNS provider
         - Log in to your domain registrar or DNS hosting provider.

>          - Create a new **CNAME** record for your desired subdomain (e.g., `mail`).

Reviewer 2: - Create a CNAME record pointing your desired subdomain (e.g., `mail`) to the odoo subdomain (for example, `projectname.odoo.com.`).

Or something similar; the details must be the same as for Odoo Online, otherwise customers won't know what value to use for the CNAME.

      - Add the alias domain inside your database

         - Log in to the database and open the :guilabel:`Settings` app.
         - In the :guilabel:`Alias Domain` field, enter the subdomain you configured (for
           example, `mail.mydomain.com`).

         .. raw:: html

            <div style="margin-top: 1rem;">

         .. image:: email_servers_inbound/alias-domain.png
            :alt: Adding an alias domain in the settings.
            :width: 300px

         .. raw:: html

              </div>

      .. spoiler:: Click here to see an example of a configuration on the Odoo portal

         In the :guilabel:`Domain Names` section, the subdomain should appear as follows:

         .. image:: email_servers_inbound/DN-portal-config-example.png
            :alt: Example of domain name records configuration in the Odoo portal.

         In the :guilabel:`DNS` section, the records should appear as follows:

         .. image:: email_servers_inbound/DNS-portal-config-example.png
            :alt: Example of a DNS records configuration in the Odoo portal.

   .. group-tab:: Odoo.sh

>       - Configure your DNS provider

Reviewer 3: Configure your DNS (remove the provider)

         - Log in to your domain registrar or DNS hosting provider.
         - Create a CNAME record that points your desired subdomain (for example, `mail`) to
           your Odoo subdomain (for example, `projectname.odoo.com`).

      - Add the subdomain to Odoo.sh settings

         - Log in to the Odoo.sh platform and open your project.
         - Select the production branch, go to the :guilabel:`Settings` tab, and locate the
           **Custom Domains** section.
         - Enter the custom subdomain (e.g., `mail.mydomain.com`) and click
           :guilabel:`Add domain`.

>          .. raw:: html

Reviewer 3: unsure we would want this to do a whitespace.

Letting the marketing decide for this

            <div style="margin-top: 1rem;">

         .. image:: email_servers_inbound/custom-subdomain-sh.png
            :alt: Adding a custom subdomain for mail to Odoo.sh project settings.

>          .. raw:: html

Reviewer 3: same than previous. Felt pretty weird

   .. image:: email_servers_inbound/incoming-mail-server.png
      :alt: Technical schema of mailing route when using a custom domain in Odoo.

.. _email-inbound-custom-domain-dns:

.. _dns_record_setup:

DNS records
-----------

> **This setup requires advanced technical knowledge.**

Reviewer: ```suggestion
.. important::
   This setup requires advanced technical knowledge.
```

A third option is to use a dedicated subdomain (for example, `mail.mydomain.com`) and configure DNS
records for it so that email traffic for that subdomain is handled by Odoo.

With this method, instead of configuring a separate mail server, you use a dedicated subdomain
that is linked to your Odoo database through DNS configuration.
Emails sent to addresses on that subdomain are then handled by Odoo and processed in the

> corresponding database.

Reviewer: I think we could condense this as follows, to avoid some repetition ... what do you think?

```suggestion
A third option is to use a dedicated subdomain (for example, `mail.mydomain.com`) and configure its DNS
records so that emails sent to addresses on that subdomain are handled by Odoo and processed directly within your database.
```

Reviewer: I see you didn't take this feedback ... I still find there's a bit too much repetition, but I can live with it 😉

.. important::

>    For Odoo Online and Odoo.sh, CNAME-based routing is only supported when using a dedicated

Reviewer: My technical knowledge is lacking here ... so ... I guess this section covers both CNAME- and MX-based routing. I was just confused to discover the term 'CNAME-based routing' for the first time in the note. Also, is the info in the note only applicable to CNAME and not MX? I need a little help to understand this ... :upside_down_face:

Author: It does indeed cover both CNAME- and MX-based routing. We only mention CNAME in the section because it is the easiest DNS record to configure as it inherits SPF for example. We then mention MX-based routing as an alternative in the note at the end of the section.

   subdomain. It cannot be configured on your root domain (e.g., `mydomain.com`).

   Note that this subdomain will be dedicated to email traffic. If it is accessed in a web
   browser, users will be redirected to the first website configured on the database (if

>    applicable).

Reviewer: Can you explain the 'If it is accessed in a web browser ...' to me?

Author: By "access via a web browser," we simply mean that if they enter the subdomain they configured (e.g., `mail.domain.com`) into a web browser, they will be redirected to the first website they configured in Odoo, regardless of how many websites they have configured.

   Since a new subdomain is being configured for inbound mail, its DNS records must also be
   set up for outbound mail. For instructions, see the

>    :doc:`Configure DNS records to send emails in Odoo <email_domain>`.

Reviewer: ```suggestion
   Since a new subdomain is being configured for inbound mail, its :doc:`DNS records must also be set up for outbound mail <email_domain>`.
```

.. tabs::
   .. group-tab:: Odoo Online

      - Access your DNS management zone

         - If your domain is hosted with a third-party registrar (for example, GoDaddy,

>            Namecheap, Cloudflare), sign in to the registrar and open the DNS management zone

Reviewer: ```suggestion
           Namecheap, Cloudflare), log in to the registrar and open the DNS management zone
```

Just for consistency with the Odoo.sh tab

           for your domain.
         - If Odoo manages your DNS:

            - Open the `database manager <https://www.odoo.com/my/databases>`_.

>             - Click the :icon:`fa-gear` (:guilabel:`gear`) button beside the database and

Reviewer: ```suggestion
            - Click the :icon:`fa-gear` (:guilabel:`gear`) icon beside the database and
```

              choose :icon:`fa-globe` :guilabel:`Domain Names`.
            - Click :guilabel:`DNS`, then :guilabel:`Add DNS record`.

      - Create a CNAME record

         - **Name / Host:** the subdomain you want to use (for example, `mail`).

>          - **Content / Target:** your database root domain (for example, `dbname.odoo.com.`).

Reviewer: Just so I understand, by **Name / Host** and **Content / Target**, are you referring to fields the user would see in the DNS management zone of a third-party registrar / Odoo respectively?

Author: Exactly :)

      - Connect the subdomain in Odoo

         - In the `database manager <https://www.odoo.com/my/databases>`_, click the

>            :icon:`fa-gear` (:guilabel:`gear`) button for your database and select

Reviewer: ```suggestion
           :icon:`fa-gear` (:guilabel:`gear`) icon for your database and select
```

           :icon:`fa-globe` :guilabel:`Domain Names`.
         - Click :guilabel:`Use my own domain`, enter the full subdomain (for example,
           `mail.mydomain.com`). When verification succeeds,
           click :guilabel:`I confirm, it's done`.

           .. note::
            If verification fails, double-check that the CNAME target ends with a trailing dot
            (for example, `dbname.odoo.com.`), as some DNS providers require it. Also allow some
            time for DNS changes to propagate — this can take anywhere from a few minutes up to 48
            hours depending on your provider. Once propagation is complete, try verifying again.

>       - Add the alias domain inside your database

Reviewer: ```suggestion
      - Add the alias domain to the database
```

For consistency with the Odoo.sh tab

         - Log in to the database and open the :guilabel:`Settings` app.
         - In the :guilabel:`Alias Domain` field, enter the subdomain you configured (for
           example, `mail.mydomain.com`).

            .. image:: email_servers_inbound/alias-domain.png
               :alt: Adding an alias domain in the settings.
               :width: 300px

>       .. spoiler:: Click here to see an example of a configuration on the Odoo portal

Reviewer: Any idea why this defaults to center-aligned? Ideally, it should be left-aligned.

Author: Idk, maybe it is related to this:
- Outside of tabs: The spoiler renders normally as a left-aligned container (`> Title of the spoiler`).
- Inside .. tabs::: The spoiler breaks because sphinx-tabs applies styling to all elements within the tab container.
Specifically, in style.css (around line 14119):
```
.sphinx-tabs button {
    /* display: inline; */
    }
    .sphinx-tabs button:focus {
      outline: none; }
```
Unchecking **display: inline** for .sphinx-tabs button in the browser DevTools fixes the alignment and makes the spoiler display "normally", just like it does outside of tabs.

Reviewer: Thanks for looking into this, @jupe-odoo ! But I guess that means we're just stuck with this center-alignment then.

         .. image:: email_servers_inbound/DN-portal-config-example.png
            :alt: Example of domain name records configuration in the Odoo portal.

         In the :guilabel:`DNS` section, the records should appear as follows:

>          .. image:: email_servers_inbound/DNS-portal-config-example.png
>             :alt: Example of a DNS records configuration in the Odoo portal.

Reviewer: ```suggestion
         .. image:: email_servers_inbound/DNS-portal-config-example.png
            :alt: Example of a DNS records configuration in the Odoo portal.
            :scale: 50%
```

I played with some different scales and I think 50% would be best here.

         .. image:: email_servers_inbound/DNS-portal-config-example.png
            :alt: Example of a DNS records configuration in the Odoo portal.

         *The second record enables Odoo to send authenticated emails from your custom domain.

>          To learn more, see:* :doc:`Configure DNS records to send emails in Odoo <email_domain>`.

Reviewer: ```suggestion
         The second record enables Odoo to send authenticated emails from your custom domain.
         .. seealso::
            :doc:`Configure DNS records to send emails in Odoo <email_domain>`
```

Not sure why this was in italics. If it was to try to show it was referring to the point about the DNS section, I think we can just be more specific in the text. Also, we generally avoid using constructions like 'To learn more, see ...', and typically either add links to regular text or use a 'See also'.

>    .. group-tab:: Odoo.sh

Reviewer: Can we try to use the same wording as in the Odoo Online tab, where relevant? I've tried below, but maybe not always successfully 😉

      - Configure your DNS

         - Log in to your domain registrar or DNS hosting provider.
         - Create a CNAME record that points your desired subdomain (for example, `mail`) to

>            your Odoo subdomain (for example, `projectname.odoo.com`).

Reviewer: Could this be presented the same way as for Odoo Online?

```suggestion
         - Create a CNAME record 
            - ......: the subdomain you want to use (for example, `mail`)
            - ......: your Odoo subdomain (for example, `projectname.odoo.com`).
```

      - Add the subdomain to Odoo.sh settings

         - Log in to the Odoo.sh platform and open your project.
         - Select the production branch, go to the :guilabel:`Settings` tab, and locate the

>            **Custom Domains** section.

Reviewer: ```suggestion
           **Custom domains** section.
```

Just going by the screenshot ...

>          - Enter the custom subdomain (e.g., `mail.mydomain.com`) and click

Reviewer: ```suggestion
         - Enter the full subdomain (e.g., `mail.mydomain.com`) and click
```

           :guilabel:`Add domain`.

            .. image:: email_servers_inbound/custom-subdomain-sh.png
               :alt: Adding a custom subdomain for mail to Odoo.sh project settings.

>       - Finally, add the alias domain to the database

Reviewer: ```suggestion
      - Add the alias domain to the database
```

      - Add the alias domain to the database

>          - Log in to the database and open :guilabel:`Settings` app.

Reviewer: ```suggestion
         - Log in to the database and open the :guilabel:`Settings` app.
```

Sorry, missed that one just now!

>          - Go to your database and open :guilabel:`Settings`.

Reviewer: Can we use the same wording as for Odoo Online, i.e.,

```suggestion
         - Log in to the database and open the :guilabel:`Settings` app.
```

>            `mail.mydomain.com`.

Reviewer: ```suggestion
         - In the :guilabel:`Alias Domain` field, enter the subdomain you configured (for example,
           `mail.mydomain.com`).
```

To use same wording as in Odoo Online tab.

            .. image:: email_servers_inbound/alias-domain.png
               :alt: Adding an alias domain in the settings.
               :width: 300px

.. note::

>    Instead of a CNAME record, you can alternatively create an **MX record** for your subdomain

Reviewer: ```suggestion
   Instead of a CNAME record, you can alternatively create an MX record for your subdomain
```
